test/tesla_test.exs

```diff
@@ -273,4 +273,34 @@ defmodule TeslaTest do
       assert get_header(env, "server") == "Cowboy"
     end
   end
+
+  describe "build_url/2" do
+    setup do
+      {:ok, url: "http://api.example.com"}
+    end
+
+    test "returns URL with query params from keyword list", %{url: url} do
+      query_params = [{:user, 3}, {:page, 2}]
+      assert build_url(url, query_params) === url <> "?user=3&page=2"
+    end
+
+    test "returns URL with new query params concated from keyword list", %{url: url} do
+      url_with_param = url <> "?user=4"
+      query_params = [{:page, 2}, {:status, true}]
```

Comment:
```suggestion
      query_params = [page: 2, status: true]
```

Reply:
Wow, I didn't realize that having a list of tuples could be the same as having a list of keywords, thanks!